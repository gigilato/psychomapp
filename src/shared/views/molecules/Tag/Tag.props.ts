import { StyleProp, ViewStyle } from 'react-native'

import { IBoxProps, IPressableProps } from '$atoms'

export interface ITagContainerProps extends IBoxProps {
  gap: number
}

export interface ITagProps extends Omit<IPressableProps, 'ID'> {
  tag: string
  backgroundColor: string
  gapStyle?: StyleProp<ViewStyle>
}
