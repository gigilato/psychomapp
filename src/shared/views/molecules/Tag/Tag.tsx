import clsx from 'clsx'
import React from 'react'

import { HStack, Pressable, Text } from '$atoms'

import { ITagContainerProps, ITagProps } from './Tag.props'

export const TagContainer = ({ gap, className, style, ...props }: ITagContainerProps) => {
  return (
    <HStack
      className={clsx('flex-wrap', className)}
      style={[{ marginTop: -gap, marginLeft: -gap }, style]}
      {...props}
    >
      {React.Children.map(props.children, (child) => {
        if (!child) return null
        return React.cloneElement<ITagProps>(
          child as React.ReactElement,
          { gapStyle: { marginTop: gap, marginLeft: gap } },
          null
        )
      })}
    </HStack>
  )
}

export const Tag = ({
  tag,
  backgroundColor,
  className,
  onPress,
  gapStyle,
  style,
  ...props
}: ITagProps) => {
  return (
    <Pressable
      ID={`tag/${tag}`}
      className={clsx('rounded-full p-xxs flex-row self-start', className)}
      style={[{ backgroundColor }, gapStyle, style]}
      disabled={!onPress}
      onPress={onPress}
      {...props}
    >
      <Text variant="smallBody" className="text-white-classic">
        {tag}
      </Text>
    </Pressable>
  )
}
