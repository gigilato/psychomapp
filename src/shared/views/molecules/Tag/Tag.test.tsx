import { render } from '@testing-library/react-native'

import { Tag, TagContainer } from './Tag'

const mockTags = [
  { tag: 'tag1', backgroundColor: '#ff00ff' },
  { tag: 'tag2', backgroundColor: '#4f00ff' },
  { tag: 'tag3', backgroundColor: '#f0007f' },
]

describe('Tag', () => {
  it('should render correctly', async () => {
    const Component = render(
      <TagContainer gap={8}>
        {mockTags.map((tag) => (
          <Tag key={tag.tag} {...tag} />
        ))}
      </TagContainer>
    )

    expect(Component).toMatchSnapshot()
  })
})
