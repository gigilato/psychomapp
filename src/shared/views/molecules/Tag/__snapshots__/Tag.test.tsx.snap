// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag should render correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "flexWrap": "wrap",
      },
      [
        undefined,
        [
          {
            "marginLeft": -8,
            "marginTop": -8,
          },
          undefined,
        ],
      ],
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "borderBottomLeftRadius": 9999,
          "borderBottomRightRadius": 9999,
          "borderTopLeftRadius": 9999,
          "borderTopRightRadius": 9999,
        },
        {
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
        },
        {
          "flexDirection": "row",
        },
        {
          "alignSelf": "flex-start",
        },
        [
          {
            "opacity": undefined,
          },
          [
            {
              "backgroundColor": "#ff00ff",
            },
            {
              "marginLeft": 8,
              "marginTop": 8,
            },
            undefined,
          ],
        ],
      ]
    }
    testID="tag/tag1"
  >
    <Text
      style={
        [
          {
            "fontFamily": "SpaceGrotesk-light",
          },
          {
            "fontSize": 12,
          },
          {
            "lineHeight": 16,
          },
          {
            "color": "#FFFCF9",
          },
        ]
      }
    >
      tag1
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "borderBottomLeftRadius": 9999,
          "borderBottomRightRadius": 9999,
          "borderTopLeftRadius": 9999,
          "borderTopRightRadius": 9999,
        },
        {
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
        },
        {
          "flexDirection": "row",
        },
        {
          "alignSelf": "flex-start",
        },
        [
          {
            "opacity": undefined,
          },
          [
            {
              "backgroundColor": "#4f00ff",
            },
            {
              "marginLeft": 8,
              "marginTop": 8,
            },
            undefined,
          ],
        ],
      ]
    }
    testID="tag/tag2"
  >
    <Text
      style={
        [
          {
            "fontFamily": "SpaceGrotesk-light",
          },
          {
            "fontSize": 12,
          },
          {
            "lineHeight": 16,
          },
          {
            "color": "#FFFCF9",
          },
        ]
      }
    >
      tag2
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "borderBottomLeftRadius": 9999,
          "borderBottomRightRadius": 9999,
          "borderTopLeftRadius": 9999,
          "borderTopRightRadius": 9999,
        },
        {
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
        },
        {
          "flexDirection": "row",
        },
        {
          "alignSelf": "flex-start",
        },
        [
          {
            "opacity": undefined,
          },
          [
            {
              "backgroundColor": "#f0007f",
            },
            {
              "marginLeft": 8,
              "marginTop": 8,
            },
            undefined,
          ],
        ],
      ]
    }
    testID="tag/tag3"
  >
    <Text
      style={
        [
          {
            "fontFamily": "SpaceGrotesk-light",
          },
          {
            "fontSize": 12,
          },
          {
            "lineHeight": 16,
          },
          {
            "color": "#FFFCF9",
          },
        ]
      }
    >
      tag3
    </Text>
  </View>
</View>
`;
